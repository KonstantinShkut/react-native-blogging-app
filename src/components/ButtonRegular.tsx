import React from 'react';
import { Button } from 'react-native-elements';
import { colorPrimary } from '../styles';
import { StyleSheet } from 'react-native';

interface Props {
  title: string;
  color?: string;
  onPress: () => any;
}

export const ButtonRegular = ({ title, color, onPress }: Props) => {
  return (
    <Button
      title={title}
      buttonStyle={{
        backgroundColor: color || colorPrimary,
      }}
      containerStyle={styles.containerStyle}
      onPress={onPress}
    />
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    borderRadius: 10,
  },
});
