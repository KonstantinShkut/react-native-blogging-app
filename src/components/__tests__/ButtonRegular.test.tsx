import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonRegular } from '../ButtonRegular';

describe('ButtonRegular', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <ButtonRegular title="title" onPress={() => {}} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with custom color', () => {
    const component = renderer.create(
      <ButtonRegular title="title" color="white" onPress={() => {}} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const component = renderer.create(
      <ButtonRegular title="title" disabled={true} onPress={() => {}} />,
    );
    expect(component).toMatchSnapshot();
  });
});
