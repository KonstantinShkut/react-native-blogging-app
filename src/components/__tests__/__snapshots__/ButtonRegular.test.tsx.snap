// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRegular renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 10,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2196F3",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        title
      </Text>
    </View>
  </View>
</View>
`;

exports[`ButtonRegular renders correctly when disabled 1`] = `
<View
  style={
    Object {
      "borderRadius": 10,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": true,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "hsl(208, 8%, 90%)",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": Object {
              "color": Array [
                208,
                8,
                63,
              ],
              "model": "hsl",
              "valpha": 1,
            },
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        title
      </Text>
    </View>
  </View>
</View>
`;

exports[`ButtonRegular renders correctly with custom color 1`] = `
<View
  style={
    Object {
      "borderRadius": 10,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        title
      </Text>
    </View>
  </View>
</View>
`;
